### PR TITLE
Document lxc exec -- args

### PR DESCRIPTION
**Pull request by jpic**

### CODE REVIEW

**Comment by brauner** [comment]
Hi, thanks for the PR. Unfortunately you forgot to sign-off your commit which is something we require (See https://github.com/lxc/lxd/blob/master/CONTRIBUTING.md.)

File: lxc/exec.go [hunk 1]
```diff
@@ -45,7 +45,7 @@ func (c *execCmd) usage() string {
 	return i18n.G(
 		`Execute the specified command in a container.
 
-lxc exec [remote:]container [--mode=auto|interactive|non-interactive] [--env EDITOR=/usr/bin/vim]... <command>
+lxc exec [remote:]container [--mode=auto|interactive|non-interactive] [--env EDITOR=/usr/bin/vim]... <command> [-- <args>]
```
Comment (jpic) [inline lxc/exec.go]: Both work, but then we could just have `-- <command line>` in the help text ?

Comment (tych0) [inline lxc/exec.go]: Both work, but they mean different things. I'd be fine with `[--] <command line>`; but I do think it's worth noting that the -- is optional.

**Comment by jpic** [comment]
It is optionnal if you don't need arguments​, this leaves us with:

    ( <command> [-- <args>] | <command> <args> )

Or:

    ( <command> [-- <args>] | <command line> )

**Comment by tych0** [comment]
On Mon, Sep 19, 2016 at 07:19:45AM -0700, James Pic wrote:
> It is optionnal if you don't need arguments​, this leaves us with:
> 
>     ( <command> [-- <args>] | <command> <args> )

This assumes your binary doesn't start with a `-`, though.

> Or:
> 
>     ( <command> [-- <args>] | <command line> )

I really don't understand this one :)

**Comment by jpic** [comment]
My bad, the second notation here doesn't work:

     ( <command> [-- <args>] | <command> <args> )

Should have been:

     ( <command> [-- <args>] | -- <command> <args> )

Should we go for that one ?

**Comment by tych0** [comment]
On Mon, Sep 19, 2016 at 07:36:52AM -0700, James Pic wrote:
> My bad, the second notation here doesn't work:
> 
>      ( <command> [-- <args>] | <command> <args> )
> 
> Should have been:
> 
>      ( <command> [-- <args>] | -- <command> <args> )
> 
> Should we go for that one ?

Why not just `[--] <command line>`? That's much easier to read IMO.

**Comment by techtonik** [comment]
> Why not just `[--] <command line>`? That's much easier to read IMO.

+1

**Comment by jpic** [comment]
`[--] <command line>` doesn't seem valid to me, because if i understand it correctly it makes the double dash optional to pass a command line, ie.:

    $ lxc exec test-trusty bash --version
    error: flag provided but not defined: --version

The only valid syntaxes are `bash -- --version` or `-- bash --version`.

**Comment by tych0** [comment]
On Tue, Sep 20, 2016 at 04:02:52AM -0700, James Pic wrote:
> `[--] <command line>` doesn't seem valid to me, because if i understand it correctly it makes the double dash optional to pass a command line, ie.:

You only need to pass it if you're passing --args to the command,
i.e.:

lxc exec foo bash

is perfectly valid.

**Comment by jpic** [comment]
Yes, i understand, the following syntaxes work:

lxc exec foo bash
lxc exec foo bash -- --version
lxc exec foo -- bash

These fail:

lxc exec foo bash --version
lxc exec foo 'bash --version'

So, I'm not sure what's the best here, I suggest <command> [-- <args>]
because this is enough to clue the user about the requirement for a double
dash if they want to pass arguments.

Would that work for you ?

**Comment by tych0** [comment]
On Tue, Sep 20, 2016 at 08:21:08AM -0700, James Pic wrote:
> Yes, i understand, the following syntaxes work:
> 
> lxc exec foo bash
> lxc exec foo bash -- --version
> lxc exec foo -- bash
> 
> These fail:
> 
> lxc exec foo bash --version
> lxc exec foo 'bash --version'
> 
> So, I'm not sure what's the best here, I suggest <command> [-- <args>]
> because this is enough to clue the user about the requirement for a double
> dash if they want to pass arguments.
> 
> Would that work for you ?

No, but `[--] <command line>` would.

**Comment by techtonik** [comment]
> So, I'm not sure what's the best here, I suggest <command> [-- <args>]

A/B testing will help. =) From my experience users compile such specifications in a very different way, so it is better concentrate on readability and user experience than on formal proof. Remember the length of email regexp that validates everything right - it is unreadable.

**Comment by stgraber** [comment]
This pull request is missing a matching .pot change. Run "make i18n" and commit the result please.

**Comment by stgraber** [comment]
Closing in favor of #2415

**Pull request closed without merging**